src/gluonts/nursery/autogluon_tabular/estimator.py

```diff
@@ -127,13 +132,49 @@ def train(
             )
             for entry in training_data
         ]
-        df = pd.concat(dfs)
+        if validation_data is not None or last_k_for_val is not None:
+            self.kwargs["auto_stack"] = False
+            logger.warning(
+                "Auto Stacking is turned off "
+                "as validation dataset is provided before input into Tabular Predictor."
+            )
+        if last_k_for_val is not None:
+            logger.log(
+                20,
+                f"last_k_for_val is provided, choosing last {last_k_for_val} of each time series as validation set.",
+            )
+            train_dfs = [tmp_df.iloc[:-last_k_for_val, :] for tmp_df in dfs]
+            validation_dfs = [
+                tmp_df.iloc[-last_k_for_val:, :] for tmp_df in dfs
+            ]
+            train_df = pd.concat(train_dfs)
+            val_df = pd.concat(validation_dfs)
+        elif validation_data is not None:
+            logger.log(20, "Validation dataset is directly provided.")
+            validation_dfs = [
+                get_features_dataframe(
+                    series=self.scaling(to_pandas(entry))[0],
+                    time_features=self.time_features,
+                    lag_indices=self.lag_indices,
+                )
+                for entry in validation_data
+            ]
+            train_df = pd.concat(dfs)
+            val_df = pd.concat(validation_dfs)
```

Comment:
This way, `validation_data` is ignored if `last_k_for_val` is set. Maybe the other way around would be better: if `validation_data` is provided, then use it, otherwise check whether `last_k_for_val` is set: I think this is less of a "surprise" if one explicitly gives validation data.